sound/soc/intel/boards/bxt_pcm512x.c

```diff
@@ -53,6 +54,19 @@ static int broxton_hdmi_init(struct snd_soc_pcm_runtime *rtd)
 
 	list_add_tail(&pcm->head, &ctx->hdmi_pcm_list);
 
+	/*
+	 * Setup the device_link to make sure that hdmi codec power is
+	 * turned on before the consumer starts to operate on the
+	 * codec register.
+	 */
+	if (!ctx->link)
+		ctx->link = device_link_add(rtd->card->dev, dai->dev,
+					    DL_FLAG_RPM_ACTIVE);
```

Comment:
Wondering, wouldn't adding a DL_FLAG_AUTOREMOVE_CONSUMER flag would eliminate the need for bxt_pcm512x_remove() below? Or maybe even also add DL_FLAG_AUTOREMOVE_SUPPLIER?

Reply:
@lyakh  I checked the DL_FLAG_AUTOREMOVE_xxx before. I found very little drivers are using these flags. What's more, looking into the code, what I understand is that we still need call some functions to release the link even we use the flags? Based on the result, I didn't add these flags for safe. And it seems Takashi is OK for using the flag DL_FLAG_RPM_ACTIVE.